code/modules/reagents/reagent_containers/glass.dm

```diff
@@ -299,11 +299,18 @@
 	else
 		return ..()
 
+/obj/item/reagent_containers/glass/bucket/afterattack()
+	.=..()
+	update_icon()
+
 /obj/item/reagent_containers/glass/bucket/update_icon()
 	cut_overlays()
 	if (!is_open_container())
 		var/image/lid = image(icon, src, "lid_[initial(icon_state)]")
 		add_overlay(lid)
+	if (is_open_container() && reagents.total_volume > 100)
+		var/image/water = image(icon, src, "water_[initial(icon_state)]")
+		add_overlay(water)
```

Comment:
does this overlay work with the wooden version below?

Reply:
if not, please either make it work, or make sure its not applied

Reply:
Yeah it should work globally.

Reply:
Being that the wood and standstone buckets are still both subtypes of glass/bucket.

Reply:
Whoops I stand corrected. Apparently it's not asserting. Easy fix.

Reply:
my main concern is that the overlay does not fit visually into the bucket